core/unit_test/TestExecutionEnvironmentNonInitializedOrFinalized.cpp

```diff
@@ -41,21 +41,102 @@ TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest,
     Kokkos::View<int> v0;
     Kokkos::View<float*> v1;
     Kokkos::View<NonTrivial**> v2;
+    return std::make_tuple(v0, v1, v2);
   };
   EXPECT_EXIT(
       {
-        make_views();
+        { auto views = make_views(); }
         std::exit(EXIT_SUCCESS);
       },
       ::testing::ExitedWithCode(EXIT_SUCCESS), "");
   EXPECT_EXIT(
       {
-        Kokkos::initialize();
-        Kokkos::finalize();
-        make_views();
+        {
+          Kokkos::initialize();
+          auto views =
+              make_views();  // views outlive the Kokkos execution environment
+          Kokkos::finalize();
+        }
+        std::exit(EXIT_SUCCESS);
+      },
+      ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::initialize();
+          Kokkos::finalize();
+          auto views = make_views();
+        }
+        std::exit(EXIT_SUCCESS);
+      },
+      ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+}
+
+TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest, views) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::View<int*> v;
+          Kokkos::initialize();
+          v = Kokkos::View<int*>("v", 10);
+          v = Kokkos::View<int*>();
+          Kokkos::finalize();
+        }
+        std::exit(EXIT_SUCCESS);
+      },
+      ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::initialize();
+          Kokkos::View<int*> v("v", 10);
+          v = {};  // assign default constructed view
+          Kokkos::finalize();
+        }
         std::exit(EXIT_SUCCESS);
       },
       ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+  EXPECT_DEATH(
+      {
+        Kokkos::initialize();
+        Kokkos::View<int*> v("v", 0);
+        Kokkos::finalize();
+      },
+      "Kokkos allocation \"v\" is being deallocated after Kokkos::finalize was "
+      "called");
+  [[maybe_unused]] std::string error_constructing_view_with_unitialized_exec =
+      "Constructing View and initializing data with uninitialized execution "
+      "space";
+  [[maybe_unused]] std::string error_constructing_exec_space_instance =
+      std::string("Kokkos::") +
+#ifdef KOKKOS_ENABLE_OPENACC
+      "Experimental::" +
+#endif
+      Kokkos::DefaultExecutionSpace::name() +
+      "::" + Kokkos::DefaultExecutionSpace::name() +
```

Comment:
Is this a mistake? `Kokkos::DefaultExecutionSpace::name()` appears twice here.

Reply:
That's how the errors that we emit look like.  See https://github.com/kokkos/kokkos/pull/7653#discussion_r1906163968

I think the intent was to expressed that it occurs in the constructor